storage/reads/array_cursor.gen.go.tmpldata

```diff
@@ -5,19 +5,37 @@
 		"Type":"float64",
 		"Aggs": [
 			{
-		 		"Name":"Count",
-				"AccType":"int64",
-				"AccTypeName":"Integer",
-		 		"AccInit":"0",
-				"Accumulate":"acc++"
-		 	},
+				"Name":"Count",
+				"OutputTypeName":"Integer",
+				"AccDecls":"var acc int64 = 0",
+				"Accumulate":"acc++",
+				"AccEmit": "c.res.Timestamps[pos] = windowEnd; c.res.Values[pos] = acc",
+				"AccReset":"acc = 0"
+			},
 			{
-		 		"Name":"Sum",
-				"AccType":"float64",
-				"AccTypeName":"Float",
-		 		"AccInit":"0",
-				"Accumulate":"acc += a.Values[rowIdx]"
-		 	}
+				"Name":"Sum",
+				"OutputTypeName":"Float",
+				"AccDecls":"var acc float64 = 0",
+				"Accumulate":"acc += a.Values[rowIdx]",
+				"AccEmit":"c.res.Timestamps[pos] = windowEnd; c.res.Values[pos] = acc",
+				"AccReset":"acc = 0"
+			},
+			{
+				"Name":"Min",
+				"OutputTypeName":"Float",
+				"AccDecls":"var acc float64 = math.MaxFloat64; var tsAcc int64",
+				"Accumulate":"if !windowHasPoints || a.Values[rowIdx] < acc { acc = a.Values[rowIdx]; tsAcc = a.Timestamps[rowIdx] }",
+				"AccEmit":"c.res.Timestamps[pos] = tsAcc; c.res.Values[pos] = acc",
+				"AccReset":"acc = math.MaxFloat64"
```

Comment:
Does `windowHasPoints` need to be reset too? Also, does it need to be initialized?

Reply:
No, `windowHasPoints` is defined in the template for all the aggregates. I understand that it's confusing that some variables are defined by the code snippets for a particular aggregate, and some are in the template.